pdu/ShortMessage.go

```diff
@@ -65,9 +80,21 @@ func (c *ShortMessage) SetMessageWithEncoding(message string, enc data.Encoding)
 func (c *ShortMessage) SetLongMessageWithEnc(message string, enc data.Encoding) (err error) {
 	c.message = message
 	c.enc = enc
+	c.dataCoding = enc.DataCoding()
 	return
 }
 
+// SetDataCoding set ShortMessage's data coding and associated Encoding.
+func (c *ShortMessage) SetDataCoding(coding byte) {
```

Comment:
We would like to avoid exposing internal bytes in each PDU as much as possible, so explicit set function is not prefered. The encoding interface should perform this for you instead.